pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go

```diff
@@ -36,18 +40,28 @@ var _ = Describe("TKGS - Create workload cluster use cases", func() {
 				clusterOptions.ClusterConfigFile = createClusterConfigFile(e2eConfig)
 			})
 			AfterEach(func() {
+				clusterOptions.CniType = cniAntrea
 				defer os.Remove(clusterOptions.ClusterConfigFile)
 			})
+
 			When("cluster class cli feature flag (features.global.package-based-lcm-beta) is set to true", func() {
 				BeforeEach(func() {
 					//set the cli feature flag as true -  (features.global.package-based-lcm-beta)
 					Expect(framework.SetCliConfigFlag(CLI_CLUSTERCLASS_FLAG, "true")).To(Succeed(), "error while setting CLI ClusterClass flag")
 					tkgctlClient, err = tkgctl.New(tkgctlOptions)
 					Expect(err).To(BeNil())
 				})
-				It("should create TKC workload cluster and delete it", func() {
+
```

Comment:
Good point. I think it is because the cluster is not supposed to be created successfully when the feature is off - as you can see in [`createClusterClassBasedClusterTest` fucntion](https://github.com/HL-EverGreen/tanzu-framework/blob/lhaoran%2Fadd-cni-tests-for-tkgs/pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go#L219) and [`createLegacyClusterTest` function](https://github.com/HL-EverGreen/tanzu-framework/blob/lhaoran%2Fadd-cni-tests-for-tkgs/pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go#L238), the tests expect cluster to fail in creation. So Shyaam suggested me to remove the tests for Calico CNI in these cases.

Reply:
To be clear, I mean the following case. Why we's need specifying ````features.global.package-based-lcm-beta``` flag for tkgs?
```When("cluster class cli feature flag (features.global.package-based-lcm-beta) is set to false"```

Reply:
Hi @maralavi , could you elaborate more about your question? Are you asking why we need to switch the feature on and off during the E2E tests? (to be honest, I don't understand either..= =

Reply:
I honestly don't know myself what ```features.global.package-based-lcm-beta``` flag is meant o control at this point and whether it applies to tkgs at all as well or not.

Reply:
Yeah I don't understand either.. Based on the test it seems the workload cluster can't be created successfully when the feature is off. Maybe serve as a negative path for the E2E tests

Reply:
Not it does get created successfully. See [here](https://github.com/vmware-tanzu/tanzu-framework/blob/a0e2c6d2087d33ad51e207af8b91161d4323f8ea/pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go#L76) for example

Reply:
Emm, interesting. I thought it would go into the `else` path [here](https://github.com/vmware-tanzu/tanzu-framework/blob/a0e2c6d2087d33ad51e207af8b91161d4323f8ea/pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go#L235). I will double check it and add CNI calico tests if the cluster is still supposed to created successfully when feature is off.

Reply:
Thanks @maralavi for catching this! After discussing with Chandra, it seems this feature flag only affects classy cluster tests. For legacy cluster tests, they are only verifying that the workload cluster can be created **irrespective** of this flag.

Since the execution path of legacy cluster tests are actually the same when this feature is on/off, I think only verifying cluster with Calico CNI when the feature is on is enough.